contracts/smart-account/interfaces/modules/SessionKeyManagers/ISessionKeyManagerModuleHybrid.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/* solhint-disable no-empty-blocks*/
+
+/**
+ * @title Session Key Manager module for Biconomy Modular Smart Accounts.
+ * @dev Stores the Session Information explicity in the storage, instead of maintainting
+ *      a merkle tree.
+ *      This reduces the amount of calldata required to validate a session key, making it cheaper on
+ *      L2s.
+ *      Allows for a session to be enabled explicity, or being batched with the first usage of said session
+ * @author Ankur Dubey - <ankur@biconomy.io>
+ */
+interface ISessionKeyManagerModuleHybrid {
+    struct SessionData {
+        uint48 validUntil;
+        uint48 validAfter;
+        address sessionValidationModule;
+        bytes sessionKeyData;
+    }
+
+    event SessionCreated(
+        address indexed sa,
+        bytes32 indexed sessionDataDigest,
+        SessionData data
```

Comment:
would there be any benefit of this?
instead can emit skm address and on skm maybe emit specific event per choice of dev.

Reply:
skm address + valid until + valid after separate

Reply:
It's for cheaply storing the session data corresponding to a sessionId on-chain, and primarily for analytics which Shantanu has been requesting.
This will also allow us to track useful metrics such as session usage per session enabled, etc.

> instead can emit skm address and on skm maybe emit specific event per choice of dev.
why would we want to emit SKM address?

Reply:
just like paymaster address is key component of AA tx. with every session key tx a skm will be involved and makes sense to emit it's address. 

My concern was emitting entire session data and then decoding it may not be straight forward

Reply:
I think decoding should be easy since it can be done directly with ABI, no?